lib/core/src/Cardano/Wallet/Primitive/Types/TokenMap/Gen.hs

```diff
@@ -58,22 +57,7 @@ import qualified Cardano.Wallet.Primitive.Types.TokenMap as TokenMap
 --------------------------------------------------------------------------------
 
 genAssetId :: Gen AssetId
-genAssetId = sized $ \size -> do
-    -- Ideally, we want to choose asset identifiers from a range that scales
-    -- /linearly/ with the size parameter.
-    --
-    -- However, since each asset identifier has /two/ components that are
-    -- generated /separately/, naively combining the generators for these two
-    -- components will give rise to a range of asset identifiers that scales
-    -- /quadratically/ with the size parameter, which is /not/ what we want.
-    --
-    -- Therefore, we pass each individual generator a size parameter that
-    -- is the square root of the original.
-    --
-    let sizeSquareRoot = max 1 $ ceiling $ sqrt $ fromIntegral @Int @Double size
-    AssetId
-        <$> resize sizeSquareRoot genTokenPolicyId
-        <*> resize sizeSquareRoot genTokenName
+genAssetId = genSized2With AssetId genTokenPolicyId genTokenName
```

Comment:
Ha nice!

Reply:
Thanks!

If we ever have to make a generator for values composed of more than two parts, we can easily make a similar `genSized3`/`genSizedWith3` function.

Though I'm hoping we won't need such a thing. 😄